lib/calendar.ts

```diff
@@ -1721,6 +1787,10 @@ abstract class GregorianBaseHelperFixedEpoch extends HelperBase {
   maximumMonthLength(calendarDate: CalendarYM): number {
     return this.minimumMonthLength(calendarDate);
   }
+  maxLengthOfMonthCodeInAnyYear(monthCode: string) {
+    const month = +ES.Call(StringPrototypeSlice, monthCode, [1]);
+    return [undefined, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][month] as number;
```

Comment:
Is the `undefined` case needed?

```suggestion
    return [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][month-1];
```

Also I could imagine DRY-ing both these lines, e.g.

```
this.getArrayValueFromMonthCode(monthCode, [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]);
```